plugins/factoids.py

```diff
@@ -107,18 +113,42 @@ def remember(text, nick, db, chan, notice, event):
     add_factoid(db, word, chan, data, nick)
 
 
+def remove_fact(chan, name, db, notice):
+    data = factoid_cache[chan].get(name.lower())
+
+    if data:
+        del_factoid(db, chan, name)
+        notice(
+            "{!r} has been forgotten, previous value was {!r}".format(
+                name, data
+            )
+        )
+    else:
+        notice("Unknown fact {!r}".format(name))
+
+
 @hook.command("f", "forget", permissions=["op", "chanop"])
 def forget(text, chan, db, notice):
-    """<word> - forgets previously remembered <word>"""
-    data = factoid_cache[chan][text.lower()]
+    """<word>... - Remove factoids with the specified names
+
+    :type text: str
+    :type chan: str
+    :type db: sqlalchemy.orm.Session
+    :type notice: function
+    """
+    for name in text.split():
```

Comment:
If possible this loop should be further down, as sending a notice per factoid is needlessly spammy

Reply:
I prefer to keep the existing behavior of saying what the value of the fact was before deleting it, which can't be done without a notice per.

Reply:
If you're removing multiple factoids, you probably know what is in them. And, you could easily map the factoids and their contents onto one line to be sent to the user.

Reply:
That wouldn't work for longer factoids, which most are.

Reply:
It already automatically splits lines that are overly long, you could also send it through a pager easily